File: Python/Warmup-2/string_splosion.py
import unittest

'''
Given a non-empty string like "Code" return a string like "CCoCodCode".


string_splosion('Code') -> 'CCoCodCode'
string_splosion('abc') -> 'aababc'
string_splosion('ab') -> 'aab'

'''

def string_splosion(str):
  
    res = ''
    x = 0
 
    for x in range(0,len(str)):  
        res = res +str[:x]
 
 
    return res + str
 
 
class Test_string_splosion(unittest.TestCase):

    def test_1(self):
        self.assertEqual(string_splosion('Code'), 'CCoCodCode')

    def test_2(self):
        self.assertEqual(string_splosion('abc'), 'aababc')

    def test_3(self):
        self.assertEqual(string_splosion('ab'), 'aab')


if __name__ == '__main__':
    unittest.main() 

File: Java/Warmup_1/stringSplosion.java
/**
 * @author 	Luis Teixeira
 *
 * @data 	5 Jun 2018
 *
 * @mail	filipe.teixeira.996@gmail.com 
 */
package Warmup_1;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

/*
 
Given a non-empty string like "Code" return a string like "CCoCodCode".


stringSplosion("Code") --> "CCoCodCode"
stringSplosion("abc") --> "aababc"
stringSplosion("ab") --> "aab"

 */
public class stringSplosion {

	private static int i = 1;
	
	@Test
	public void Test1()
	{	
		System.out.println("**********Test "+i+"**********");
		assertEquals("stringSplosion(\"Code\") --> \"CCoCodCode\"", "CCoCodCode", stringSplosion("Code"));
		i++;
	}
	
	@Test
	public void Test2()
	{	
		System.out.println("**********Test "+i+"**********");
		assertEquals("stringSplosion(\"abc\") --> \"aababc\"", "aababc", stringSplosion("abc"));
		i++;
	}
	
	@Test
	public void Test3()
	{	
		System.out.println("**********Test "+i+"**********");
		assertEquals("stringSplosion(\"ab\") --> \"aab\"", "aab", stringSplosion("ab"));
		i++;
	}
	
	public String stringSplosion(String str) {
		
		String one = str.substring(0, 1)+str.substring(0, 1);
		
		return one+str.substring(1,2)+str;
	}

}
